stdlib/jdbc/src/main/java/org/ballerinax/jdbc/statement/UpdateStatement.java

```diff
@@ -70,26 +71,36 @@ public Object execute() {
         checkAndObserveSQLAction(strand, datasource, query);
         boolean isInTransaction = strand.isInTransaction();
         String errorMessagePrefix = "failed to execute update query: ";
+        MapValue<String, Object> generatedKeys = new MapValueImpl<>();
         try {
             ArrayValue generatedParams = constructParameters(parameters);
             conn = getDatabaseConnection(strand, client, datasource);
             String processedQuery = createProcessedQueryString(query, generatedParams);
-            stmt = conn.prepareStatement(processedQuery, Statement.RETURN_GENERATED_KEYS);
+
+            if (returnGeneratedKeys) {
+                stmt = conn.prepareStatement(processedQuery, PreparedStatement.RETURN_GENERATED_KEYS);
+            } else {
+                stmt = conn.prepareStatement(processedQuery);
+            }
+
             ProcessedStatement processedStatement = new ProcessedStatement(conn, stmt, generatedParams,
                     datasource.getDatabaseProductName());
             stmt = processedStatement.prepare();
             int count = stmt.executeUpdate();
-            MapValue<String, Object> generatedKeys;
-            if (!isDdlStatement()) {
-                rs = stmt.getGeneratedKeys();
-                //This result set contains the auto generated keys.
-                if (rs.next()) {
-                    generatedKeys = getGeneratedKeys(rs);
+            if (returnGeneratedKeys) {
+                if (isKeyRetrievalSupported()) {
+                    rs = stmt.getGeneratedKeys();
+                    //This result set contains the auto generated keys.
+                    if (rs.next()) {
+                        generatedKeys = getGeneratedKeys(rs);
+                    }
                 } else {
-                    generatedKeys = new MapValueImpl<>();
+                    handleErrorOnTransaction(this.strand);
+                    String message = "The returnGeneratedKeys only support INSERT, UPDATE, DELETE and MERGE " +
+                            "operations.";
+                    checkAndObserveSQLError(strand, "execute update failed: " + message);
+                    return ErrorGenerator.getSQLApplicationError(message);
```

Comment:
Can we add a test case for this scenario as well.

Reply:
Added.